teps/0078-tekton-catalog-support-tiers.md

```diff
@@ -0,0 +1,345 @@
+ pass a list of comma separated directory patterns to this Task's
+ sparseCheckoutDirectories param.
+
+🗒  Minimum Pipeline Version: 0.21.0
+
+⭐ ️Rating: 3.7
+
+🏷 Tags
+  ∙ git
+
+⚒ Install Command:
+  tkn hub install task git-clone
+```
+
+```shell
+$ bin/tkn hub search --tier community
+
+NAME              KIND   CATALOG   TIER        DESCRIPTION                          TAGS
+git-clone (0.4)   Task   Tekton    Community   The git-clone Task will clone...     git
+```
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+ 
+- _**Simplicity**_: providing only three support tiers, and making it optional with a default value, provides a simple 
+mechanism to define support tiers for resources using annotations to surface information is consistent with other 
+available information, e.g. `tekton.dev/tags` and `tekton.dev/platforms`.
+
+- _**Reusability**_: annotations are used to define information about resources, this proposal reuses annotations for
+  the same purpose.
+
+- _**Flexibility**_: annotations can be easily extended to support additional support tiers.
+
+## Migration Strategy
+
+<!--
+Use this section to detail whether this feature needs an upgrade or
+migration strategy. This is especially useful when we modify a
+behavior or add a feature that may replace and deprecate a current one.
+-->
+
+Because the support tier annotation is optional, and defaults to *community* tier, then this proposal would provide 
+backward compatibility to old versions of resources in the Tekton Catalog and Hub. Searching and filtering for the old 
+resources based on tiers, which don't actually have tiers specified, will still return successfully with the *community* 
+tier. 
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+Add new `tekton.dev/community`, `tekton.dev/verified`, and `tekton.dev/official` annotations to the resource yaml file 
+with "true" or "false" to indicate the support tier, for example:
+
+```yaml
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: git
+    tekton.dev/community: true
+```
+
+```yaml
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: git
+    tekton.dev/verified: true
+```
+
+```yaml
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: git
+    tekton.dev/official: true
+```
+
+However, we need to add more validation to ensure that any combinations of the annotations are compatible. Moreover, for
+every new tier, we will need to add a new annotation instead of extending the acceptable fields.
+
```

Comment:
trying to brainstorm other alternatives

- we could use folder structure (con: wouldn't be possible to get the support tier info just looking at the task - ive suggested a requirement around this)
- we could use separate repositories (same con as previous)